### PR TITLE
Fix `betas` calculation on `VQD` and async. cost evaluation

### PR DESCRIPTION
**Pull request by ElePT**
### Summary
This PR fixes the `betas` auto-evaluation on `VQD` for `PauliSumOp` operators (pending deprecation), and adds support for beta auto-evaluation with `SparsePauliOp`. The code was supposed to iterate over primitive coefficients, and was instead obtaining a series of `PauliSumOp` coefficients (which were always `1.0`). 

In addition to the `betas` fix, this PR includes a change in the energy eval. method to leverage the async capabilities of the primitives (the job results are retrieved after both jobs are submitted, instead of sequentially submitting and retrieving results), as this was the initial intention.

### Details and comments
The unit tests have also been updated not to use `PauliSumOp` internally, and to test `SparsePauliOp` imports.

### CODE REVIEW

**Comment by woodsp-ibm** [comment]
This should have a reno and we probably want to release this in a bug fix release too right?

File: qiskit/algorithms/eigensolvers/vqd.py [hunk 1]
```diff
@@ -227,15 +228,19 @@ def compute_eigenvalues(
         if self.betas is None:
             if isinstance(operator, PauliSumOp):
                 upper_bound = abs(operator.coeff) * sum(
-                    abs(operation.coeff) for operation in operator
+                    abs(operation.primitive.coeffs) for operation in operator
```
Comment (Cryoris) [inline qiskit/algorithms/eigensolvers/vqd.py:231]: Could this simplified to the following?
```suggestion
                    np.abs(operator.primitive.coeffs) 
```
Or maybe even just cast to a `SparsePauliOp` for the whole function?
```
if isinstance(operator, PauliSumOp):
    operator = operator.coeff * operator.primitive
```

File: qiskit/algorithms/eigensolvers/vqd.py [hunk 2]
```diff
@@ -360,27 +366,30 @@ def _get_evaluate_energy(
...
+                # compute overlap cost
                 fidelity_job = self.fidelity.run(
                     [self.ansatz] * (step - 1),
                     prev_states,
                     [parameters] * (step - 1),
                 )
+
                 costs = fidelity_job.result().fidelities
 
                 for (state, cost) in zip(range(step - 1), costs):
```
Comment (Cryoris) [inline qiskit/algorithms/eigensolvers/vqd.py:384]: Unrelated but I was stumbling over this: Wouldn't the following be more concise? 
```suggestion
                for state, cost in enumerate(costs):
```

File: qiskit/algorithms/eigensolvers/vqd.py [hunk 3]
```diff
@@ -360,27 +366,30 @@ def _get_evaluate_energy(
...
-                    values += np.real(betas[state] * cost)
+                    total_cost += np.real(betas[state] * cost)
+
+            try:
+                estimator_result = estimator_job.result()
+                values = estimator_result.values + total_cost
```
Comment (Cryoris) [inline qiskit/algorithms/eigensolvers/vqd.py:389]: Not blocking but would it be better to take the summation out of the try block to distinguish between the estimator failing and the addition failing (due to e.g. wrong types, or the result being None,...)?

File: test/python/algorithms/eigensolvers/test_vqd.py [hunk 4]
```diff
@@ -116,7 +115,16 @@ def test_full_spectrum(self):
             result.eigenvalues.real, self.h2_energy_excited, decimal=2
         )
 
-    @data(H2_PAULI, H2_OP)
+    def test_betas_autoeval(self):
```
Comment (Cryoris) [inline test/python/algorithms/eigensolvers/test_vqd.py:118]: Is there a way we could test the autoevaluation itself? This here would also fail if something else in the algorithm is going wrong 🙂 What about e.g. checking the log?

Comment (ElePT) [reply test/python/algorithms/eigensolvers/test_vqd.py:118]: Yeah... I thought about maybe adding the betas to the metadata to be able to check via callback, it's either that or overwriting `self.betas` with the autoevaluated value.

File: qiskit/algorithms/eigensolvers/vqd.py (hunk 3 above)
Comment (ElePT) [reply qiskit/algorithms/eigensolvers/vqd.py:389]: sure

File: qiskit/algorithms/eigensolvers/vqd.py [hunk 5]
```diff
@@ -225,17 +226,21 @@ def compute_eigenvalues(
...
                 raise NotImplementedError(
-                    r"Beta autoevaluation is only supported for operators"
-                    f"of type PauliSumOp, found {type(operator)}."
-                )
+                    r"Beta autoevaluation is not supported for operators"
+                    f"of type {type(operator)}."
+                ) from exc
+
+            betas = [upper_bound * 10] * (self.k)
+            logger.info(f"beta autoevaluated to {betas[0]}")
```
Comment (Cryoris) [inline qiskit/algorithms/eigensolvers/vqd.py:243]: Pylint complains if logger messages are not lazily evaluated 🙂 
```suggestion
            logger.info("beta autoevaluated to %s", betas[0])
```

File: test/python/algorithms/eigensolvers/test_vqd.py [hunk 6]
```diff
@@ -12,7 +12,10 @@
 
 """ Test VQD """
 
+import sys
 import unittest
+import logging
+
```
Comment (Cryoris) [inline test/python/algorithms/eigensolvers/test_vqd.py:18]:
```suggestion
import unittest
```

**Review by Cryoris — APPROVED** [review]
LGTM thanks for the fix!

**Pull request merged**